# Permutations and Combinations
***
### Description
This project performs probability analysis using combinatorics methods to solve a modified experiment reminiscent of "Lady Tasting Tea". The problem uses 12 cups of tea: 6 of them with milk added first and 6 with tea added initially. The challenge is to calculate the probability that a person will correctly identify all 6 cups of milk added initially, as well as the probability of making one or two mistakes.

In [7]:
# Import lib
from math import comb

The comb function from the math module is used to compute binomial coefficients.

In [8]:
# Total number of combinations
total_combinations = comb(12, 6)
print(f"Total number of combinations: {total_combinations}")

Total number of combinations: 924


The total number of possible combinations to select 6 cups out of 12 is equale 924.
The total number of combinations shows how many possible sets can be made from 12 cups by selecting 6. This will allow to calculate probabilities for specific situations.

In [9]:
# The probability of guessing all 6 cups
favorable_combinations_all_correct = 1
probability_all_correct = favorable_combinations_all_correct / total_combinations
print(f"The probability of guessing all 6 cups: {probability_all_correct:.10f}")

The probability of guessing all 6 cups: 0.0010822511


The probability that all 6 cups will be guessed is calculated. Since there is only one combination where all the cups are correctly guessed, the probability is defined as the ratio of one favorable case to the total number of combinations.

The probability of guessing all 6 cups is very low, which is confirmed by the result. This shows that randomly guessing all the correct cups is almost impossible.

In [10]:
# Number of Ways to Guess Exactly 5 Correct Cups
ways_five_correct = comb(6, 5) * comb(6, 1)
print(f"Number of Ways to Guess Exactly 5 Correct Cups: {ways_five_correct}")

Number of Ways to Guess Exactly 5 Correct Cups: 36


It calculates the number of ways in which you can guess exactly 5 correct cups out of 6. First, choose 5 correct cups out of 6 and then add one wrong cup out of the remaining 6.

The number of ways to guess the 5 correct cups is much greater than the number of ways to guess all 6 correct ones. This makes it easier to calculate the probability for situations like this.

# End
***